tests/FilterMapTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Spatie\CollectionMacros\Test;
+
+use Illuminate\Support\Collection;
+
+class FilterMapTest extends TestCase
+{
+    /** @test */
+    public function it_returns_a_mapped_collection_without_empty_values()
+    {
+        $result = Collection::make([1, 2, 3, 4, 5, 6])->filterMap(function ($number) {
+            $half = $number / 2;
+
+            return is_float($half)
+                ? null
+                : $half;
+        });
+
+        $this->assertEquals([1, 2, 3], $result->values()->toArray());
+    }
+
+    /** @test */
+    public function it_returns_a_mapped_collection_with_a_custom_filter()
+    {
+        $result = Collection::make([1, 2, 3, 4, 5, 6])->filterMap(function ($number) {
+            return $number / 2;
+        }, function ($number) {
+            return ! is_float($number);
```

Comment:
If we check whether the given `$number` is an integer there's no need for the negation.

```php
return is_integer($number);
```